src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -746,13 +746,16 @@ public function offsetGet($option)
             }
 
             if (!$valid) {
-                $keys = array_keys($invalidTypes);
-
-                if (1 === \count($keys) && '[]' === substr($keys[0], -2)) {
-                    throw new InvalidOptionsException(sprintf('The option "%s" with value %s is expected to be of type "%s", but one of the elements is of type "%s".', $option, $this->formatValue($value), implode('" or "', $this->allowedTypes[$option]), $keys[0]));
-                }
-
-                throw new InvalidOptionsException(sprintf('The option "%s" with value %s is expected to be of type "%s", but is of type "%s".', $option, $this->formatValue($value), implode('" or "', $this->allowedTypes[$option]), implode('|', array_keys($invalidTypes))));
+                throw new InvalidOptionsException(
+                    sprintf(
+                        'The option "%s" with value %s is expected to be of type "%s", but %s of type "%s".',
+                        $option,
+                        $this->formatValue($value),
+                        implode('" or "', $this->allowedTypes[$option]),
+                        \is_array($value) && '[]' === substr($allowedTypes[0], -2) ? 'one of the elements is' : 'is',
```

Comment:
Could you please add this use case `[[123], 'string']? I afraid it's still not accurate.